src/crystal/system/win32/visual_studio.cr

```diff
@@ -0,0 +1,55 @@
+require "json"
+
+module Crystal::System::VisualStudio
+  struct Installation
+    include JSON::Serializable
+
+    @[JSON::Field(key: "installationPath")]
+    getter directory : String
+
+    @[JSON::Field(key: "installationVersion")]
+    getter version : String
+
+    # unused fields not mapped
+  end
+
+  def self.find_latest_msvc_path : String?
+    # ported from https://github.com/microsoft/vswhere/wiki/Find-VC
+    if vs_installations = get_vs_installations
+      vs_installations.sort_by! &.version
+      vs_installations.reverse_each do |installation|
+        version_path = "#{installation.directory}\\VC\\Auxiliary\\Build\\Microsoft.VCToolsVersion.default.txt"
+        next unless ::File.exists?(version_path)
+
+        version = ::File.read(version_path).chomp
+        next if version.empty?
+
+        return ::Path["#{installation.directory}\\VC\\Tools\\MSVC\\#{version}"].normalize.to_s
```

Comment:
Style: I don't see why backslashes should be hardcoded, even if it is currently very Windows-specific.
Doing it this way also eliminates potential extra slashes on either side.

```suggestion
        return ::Path[installation.directory, "VC", "Tools", "MSVC", version].to_s
```

Here and elsewhere.

Or wait, what's the difference from

```suggestion
        return ::File.join(installation.directory, "VC", "Tools", "MSVC", version)
```